packages/block-library/src/image/index.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Server-side rendering of the `core/image` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Adds a data-id attribute to the core Image block when nested in a Gallery block.
+ *
+ * @param  WP_Post $post The WP post.
+ * @return string        Returns the post content with the data-id attribute added to gallery images.
+ */
+function get_block_core_image_post_content( $post ) {
+	if ( is_admin() ) {
```

Comment:
```suggestion
	if ( is_admin() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
```

Is there are case where we'd want to run this code for rest requests?

I'm thinking it's safe to return early if we're targeting post data on the frontend, unless I'm missing a use case.

```
	if ( is_admin() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
		return;
	}
```

For example, add a `var_dump('I LOVE THE BAND KISS!')` or some other logging after this if statement, and try to save a post. It doesn't have to contain a gallery.

The POST response will contain invalid, but possibly true, content: `string(21) "I LOVE THE BAND KISS!" {id: 115, date: "2021-11-02T03:18:06", date_gmt: "2021-11-02T03:18:06",…}`

Pinging a WP API endpoint that contains gallery post content still works for me with this check in place:

E.g., `http://localhost:4759/wp-json/wp/v2/posts/104`
<img width="1362" alt="Screen Shot 2021-11-02 at 2 05 43 pm" src="https://user-images.githubusercontent.com/6458278/139781212-86e2b5a7-bc24-4cf0-b01d-1c73e76f334a.png">

Reply:
I'd be somewhat wary about using `is_admin()` in a condition check here. All it means is that the context when this post is setup is happening without `wp-admin`, it doesn't actually mean that the post is being edited or the like. Somewhat could just as easily be expecting to get this data over an `admin-ajax.php` call or simply trying to render their gallery in an admin page.

The same goes for `REST_REQUEST`. It can both be used in the Block Editor. But it can also be used by someone making an API call to grab data for somewhere on the front-end of their website where they would expect the ids to be present.

Reply:
Thanks for the advice @TimothyBJacobs 🙇 

I think @glendaviesnz was trying to find ways to isolate the logic to frontend rendering to avoid block validation errors, given that `data-id` isn't output by the save method of the image block.

🤔 Assuming block validation wasn't an issue, do you see any issues with removing the if check here entirely?

Cheers!

Reply:
> Assuming block validation wasn't an issue, do you see any issues with removing the if check here entirely?

No, I think removing the check entirely would be ideal.